src/callbacks/liquidity/BaselineV2/BaselineAxisLaunch.sol

```diff
@@ -274,33 +295,32 @@ contract BaselineAxisLaunch is BaseCallback, Policy, Owned {
 
         // Set the ticks for the Baseline pool initially with the following assumptions:
         // - The floor range is 1 tick spacing wide
-        // - The floor range contains the active tick
-        // - The floor range lower tick is not equal to the active tick
-        // - There is no anchor (width of the anchor tick range is 0)
-        // - The discovery range is set to the active tick plus the discoveryTickWidth multiplied by the tick spacing
-        int24 activeTickUpperBoundary;
+        // - The anchor range is `anchorTickWidth` tick spacings wide, above the floor range
+        // - The discovery range is `discoveryTickWidth` tick spacings wide, above the anchor range
+        // - The anchor range contains the active tick
+        // - The anchor range upper tick is the active tick rounded up to the nearest tick spacing
+        // - The other range boundaries are calculated accordingly
         {
-            // Initially, the active tick upper boundary is the active tick rounded up
-            activeTickUpperBoundary = BPOOL.getActiveTS();
+            // Initially, the anchor range upper is the active tick rounded up
+            int24 anchorRangeUpper = _roundUpTickToSpacing(activeTick, tickSpacing);
```

Comment:
I think this is right. The only part I'm not clear on is if the initialize (active) tick must be rounded to the nearest multiply of the tick spacing instead of rounding the range upper.

Reply:
There's another issue with this (the Uniswap V3 pool is now being initialised in the BPOOL constructor) which will require some re-design, but AFAIK the UniswapV3Pool does not require the initialized/active tick/sqrtPriceX96 to be rounded to tick spacing. https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/UniswapV3Pool.sol#L271